[Backport] Fails to register java.lang.Thread methods for reflection

**Describe the issue**
It's not possible to register `java.lang.Thread`'s methods for reflection.

**Steps to reproduce the issue**
```
cat > Main.java <<EOF                                                                                                                                                                     
public class Main {                                  
    public static void main(String[] args) {                                
        System.out.println("Hello World!");     
    }                                                                  
}   
EOF
javac Main.java
cat > reflect-config.json <<EOF
[
    {
        "name": "java.lang.Thread",
        "allDeclaredMethods": true
    }
]
EOF
native-image -H:ReflectionConfigurationFiles=./reflect-config.json -cp ./ Main
```

**Describe GraalVM and your environment:**
 - GraalVM version (latest snapshot builds can be found [here](https://github.com/graalvm/graalvm-ce-dev-builds/releases)), or commit id if built from source: **GraalVM CE 21.0.2+13.1**
 - JDK major version: **21**
 - OS: **Fedora 40**
 - Architecture: **AMD64**


**More details**

The build fails with
```
[8/8] Creating image...       [*
]                                                                        (0.0s @ 0.46GB)
------------------------------------------------------------------------------------------------------------------------
                        0.5s (5.0% of total time) in 97 GCs | Peak RSS: 1.04GB | CPU load: 15.62
------------------------------------------------------------------------------------------------------------------------
Produced artifacts:
 /tmp/svm_err_b_20240912T135435.617_pid527317.md (build_info)
========================================================================================================================
Failed generating 'helloworld' after 9.8s.

The build process encountered an unexpected error:

> java.lang.RuntimeException: There was an error linking the native image: Linker command exited with 1

Linker command executed:
/usr/bin/gcc -z noexecstack -Wl,--gc-sections -Wl,--version-script,/tmp/SVM-7643345261270254996/exported_symbols.list -Wl,-x -o /tmp/helloworld helloworld.o /home/zakkak/.sdkman/candidates/java/21.0.2-graalce/lib/svm/clibraries/linux-amd64/liblibchelper.a /home/zakkak/.sdkman/candidates/java/21.0.2-graalce/lib/static/linux-amd64/glibc/libnet.a /home/zakkak/.sdkman/candidates/java/21.0.2-graalce/lib/static/linux-amd64/glibc/libnio.a /home/zakkak/.sdkman/candidates/java/21.0.2-graalce/lib/static/linux-amd64/glibc/libjava.a /home/zakkak/.sdkman/candidates/java/21.0.2-graalce/lib/static/linux-amd64/glibc/libzip.a /home/zakkak/.sdkman/candidates/java/21.0.2-graalce/lib/svm/clibraries/linux-amd64/libjvm.a -Wl,--export-dynamic -v -L/tmp/SVM-7643345261270254996 -L/home/zakkak/.sdkman/candidates/java/21.0.2-graalce/lib/static/linux-amd64/glibc -L/home/zakkak/.sdkman/candidates/java/21.0.2-graalce/lib/svm/clibraries/linux-amd64 -lz -ldl -lpthread -lrt

Linker command output:
Using built-in specs.
COLLECT_GCC=/usr/bin/gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/14/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:amdgcn-amdhsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-redhat-linux
Configured with: ../configure --enable-bootstrap --enable-languages=c,c++,fortran,objc,obj-c++,ada,go,d,m2,lto --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-shared --enable-threads=posix --enable-checking=release --enable-multilib --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-gcc-major-version-only --enable-libstdcxx-backtrace --with-libstdcxx-zoneinfo=/usr/share/zoneinfo --with-linker-hash-style=gnu --enable-plugin --enable-initfini-array --with-isl=/builddir/build/BUILD/gcc-14.2.1-20240801/obj-x86_64-redhat-linux/isl-install --enable-offload-targets=nvptx-none,amdgcn-amdhsa --enable-offload-defaulted --without-cuda-driver --enable-gnu-indirect-function --enable-cet --with-tune=generic --with-arch_32=i686 --build=x86_64-redhat-linux --with-build-config=bootstrap-lto --enable-link-serialization=1
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 14.2.1 20240801 (Red Hat 14.2.1-1) (GCC) 
COMPILER_PATH=/usr/libexec/gcc/x86_64-redhat-linux/14/:/usr/libexec/gcc/x86_64-redhat-linux/14/:/usr/libexec/gcc/x86_64-redhat-linux/:/usr/lib/gcc/x86_64-redhat-linux/14/:/usr/lib/gcc/x86_64-redhat-linux/
LIBRARY_PATH=/usr/lib/gcc/x86_64-redhat-linux/14/:/usr/lib/gcc/x86_64-redhat-linux/14/../../../../lib64/:/lib/../lib64/:/usr/lib/../lib64/:/usr/lib/gcc/x86_64-redhat-linux/14/../../../:/lib/:/usr/lib/
COLLECT_GCC_OPTIONS='-z' 'noexecstack' '-o' '/tmp/helloworld' '-v' '-L/tmp/SVM-7643345261270254996' '-L/home/zakkak/.sdkman/candidates/java/21.0.2-graalce/lib/static/linux-amd64/glibc' '-L/home/zakkak/.sdkman/candidates/java/21.0.2-graalce/lib/svm/clibraries/linux-amd64' '-mtune=generic' '-march=x86-64' '-dumpdir' '/tmp/helloworld.'
 /usr/libexec/gcc/x86_64-redhat-linux/14/collect2 -plugin /usr/libexec/gcc/x86_64-redhat-linux/14/liblto_plugin.so -plugin-opt=/usr/libexec/gcc/x86_64-redhat-linux/14/lto-wrapper -plugin-opt=-fresolution=/tmp/cckk5rty.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s --build-id --no-add-needed --eh-frame-hdr --hash-style=gnu -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -o /tmp/helloworld -z noexecstack /usr/lib/gcc/x86_64-redhat-linux/14/../../../../lib64/crt1.o /usr/lib/gcc/x86_64-redhat-linux/14/../../../../lib64/crti.o /usr/lib/gcc/x86_64-redhat-linux/14/crtbegin.o -L/tmp/SVM-7643345261270254996 -L/home/zakkak/.sdkman/candidates/java/21.0.2-graalce/lib/static/linux-amd64/glibc -L/home/zakkak/.sdkman/candidates/java/21.0.2-graalce/lib/svm/clibraries/linux-amd64 -L/usr/lib/gcc/x86_64-redhat-linux/14 -L/usr/lib/gcc/x86_64-redhat-linux/14/../../../../lib64 -L/lib/../lib64 -L/usr/lib/../lib64 -L/usr/lib/gcc/x86_64-redhat-linux/14/../../.. --gc-sections --version-script /tmp/SVM-7643345261270254996/exported_symbols.list -x helloworld.o /home/zakkak/.sdkman/candidates/java/21.0.2-graalce/lib/svm/clibraries/linux-amd64/liblibchelper.a /home/zakkak/.sdkman/candidates/java/21.0.2-graalce/lib/static/linux-amd64/glibc/libnet.a /home/zakkak/.sdkman/candidates/java/21.0.2-graalce/lib/static/linux-amd64/glibc/libnio.a /home/zakkak/.sdkman/candidates/java/21.0.2-graalce/lib/static/linux-amd64/glibc/libjava.a /home/zakkak/.sdkman/candidates/java/21.0.2-graalce/lib/static/linux-amd64/glibc/libzip.a /home/zakkak/.sdkman/candidates/java/21.0.2-graalce/lib/svm/clibraries/linux-amd64/libjvm.a --export-dynamic -lz -ldl -lpthread -lrt -lgcc --push-state --as-needed -lgcc_s --pop-state -lc -lgcc --push-state --as-needed -lgcc_s --pop-state /usr/lib/gcc/x86_64-redhat-linux/14/crtend.o /usr/lib/gcc/x86_64-redhat-linux/14/../../../../lib64/crtn.o
/usr/bin/ld: helloworld.o:(.data+0xe0): undefined reference to `Java_java_lang_Thread_getStackTrace0'
/usr/bin/ld: helloworld.o:(.data+0x3d8): undefined reference to `Java_java_lang_Thread_getNextThreadIdOffset'
collect2: error: ld returned 1 exit status
```

The issue does not manifest in GraalVM for JDK 22 due to: https://github.com/oracle/graal/commit/369f0ff8b0a1c2b1051065b8c5d96937d058c922#diff-0aac6c381ea903b15210c9b8e29f378ef804565a21270977b0dd5bdf9e826982

Also tracked in https://github.com/oracle/graal/issues/9672

--- a/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/thread/Target_java_lang_Thread.java
+++ b/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/thread/Target_java_lang_Thread.java
@@ -623,6 +623,10 @@ public final class Target_java_lang_Thread {
         return JavaThreads.getStackTrace(false, JavaThreads.fromTarget(this));
     }
 
+    @Delete
+    @SuppressWarnings({"static-method"})
+    private native Object getStackTrace0();
+
     /** @see com.oracle.svm.core.jdk.StackTraceUtils#asyncGetStackTrace */
     @Delete
     @TargetElement(onlyWith = JDK19OrLater.class)
@@ -798,6 +802,9 @@ public final class Target_java_lang_Thread {
     @Alias
     @TargetElement(onlyWith = JDK19OrLater.class)
     native long threadId();
+
+    @Delete
+    static native long getNextThreadIdOffset();
 }
 
 @TargetClass(value = Thread.class, innerClass = "Builder", onlyWith = JDK19OrLater.class)

